lib/Sema/CSDiag.cpp

```diff
@@ -2356,7 +2258,8 @@ Expr *FailureDiagnosis::typeCheckChildIndependently(
   // holding on to an expression containing open existential types but
   // no OpenExistentialExpr, which breaks invariants enforced by the
   // ASTChecker.
-  eraseOpenedExistentials(subExpr, CS);
+  if (isa<OpenExistentialExpr>(subExpr))
```

Comment:
I think this check shouldn't be done, because `subExpr` could have `OpenExistentialExpr` as child too.

Reply:
Are child `OpenExistentialExpr`s a problem, though? I thought the 'hack' here was more about preventing orphan `OpaqueValueExprs` (that trip the verifier) when subExpr itself became an `OpenExistentialExpr` but the original expression was still used by the caller.

Reply:
I think they are still a problem, because diagnostics would re-typecheck bottom-up and create variable number `OpenExistentialExpr` along the way, based on what sub-expressions where, which if discovered again by the constraint generator would assert.

Reply:
`SanitizeExpr` (run inside `generateConstraints`) now erases `OpenExistentialExpr`s though, so won't the child `OpenExistentialExpr`s  be cleaned up before the constraint generator sees them?

Sorry for all the questions, just trying to make sure I understand. We can chat in person this afternoon if that's easier!

Reply:
Ah, I think you might be right since erase is moved to sanitize we might need to care about anything embedded anymore.